api/v1alpha3/awscluster_webhook.go

```diff
@@ -39,8 +39,10 @@ func (r *AWSCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awscluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=awsclusters,versions=v1alpha3,name=validation.awscluster.infrastructure.cluster.x-k8s.io,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-awscluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=awsclusters,versions=v1alpha3,name=validation.awscluster.infrastructure.cluster.x-k8s.io,sideEffects=None
```

Comment:
Did you use `kubebuilder create webhook` for this? The name looks wrong (it still has `validation`). Also I think `sideEffects` shouldn't be `None`? cc @vincepri

Reply:
Side effect should be None, see https://github.com/kubernetes-sigs/cluster-api/issues/2851

Reply:
will update name. thanks